### ISSUE TITLE
Random Crashing Persists

### ISSUE THREAD

**Issue by Onemario1234** [Stereowalker/Survive#150]
[latest.log](https://github.com/Stereowalker/Survive/files/6873241/latest.log)

Even after updating both Survive and UnionLib to their latest versions, I still get inconsistent crashing, usually occurring after damaging mobs or destroying blocks.

**Comment by Stereowalker** [comment]
Doesn't this look like a level hearts issue?

**Comment by Stereowalker** [comment]
java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Integer
	at com.firecontroller1847.levelhearts.LevelHearts.recalcPlayerHealth(LevelHearts.java:97)
	at com.firecontroller1847.levelhearts.Events.onPlayerLevelUp(Events.java:273)

**Comment by Onemario1234** [comment]
Oh shoot, is it LevelHearts? I honestly didn't even consider that. My apologies, I'm not good with crash logs.

**Comment by Onemario1234** [comment]
That's weird though, because LevelHearts seems to work fine. What's causing the crashes?

**Comment by Onemario1234** [comment]
> That's weird though, because LevelHearts seems to work fine. What's causing the crashes?

I stand corrected. I finally realized the connection, and partially understand the part of the log you highlighted. Gaining hearts upon leveling up is what causes the crashes. Odd, but noted. I wonder why it crashes, and what connection the crashes have with Survive.

**Comment by Onemario1234** [comment]
Because these crashes never started happening until I installed Survive. LevelHearts itself normally works fine.

**Comment by Stereowalker** [comment]
Bring this up with the authors of level hearts to get their side of the story so that we can address the issue

**Comment by Onemario1234** [comment]
> Bring this up with the authors of level hearts to get their side of the story so that we can address the issue

Will do.

**Comment by Onemario1234** [comment]
The author has been notified, and will do some debugging in the next few days to figure out the issue.

**Comment by Stereowalker** [comment]
Okay

**Comment by Onemario1234** [comment]
After a bunch of debugging, we figured out the issue. It wasn't Survive, nor was it LevelHearts. It was Configured.

**Comment by FireController1847** [comment]
Hey there, this issue can now be closed. We found the issue to be on MrCrayfish's Configured mod improperly handling types and allowing invalid types to be considered valid configuration entries (causing my Integer-array to contain strings, failing to parse).

If you're interested in details because you're bored, here's my issue and his issues where we investigated.

https://gitlab.com/FireController1847/levelhearts/-/issues/37
https://github.com/MrCrayfish/Configured/issues/15
https://github.com/MrCrayfish/Configured/issues/16
